x/valset-pref/msg_server_test.go

```diff
@@ -584,7 +584,90 @@ func (suite *KeeperTestSuite) TestWithdrawDelegationRewards() {
 
 			} else {
 				suite.Require().Error(err)
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestDelegateBondedTokens() {
+	suite.SetupTest()
+
+	testLock := suite.SetupLocks(sdk.AccAddress([]byte("addr1---------------")))
+
+	tests := []struct {
```

Comment:
Can we add test case for `ForceUnlockBondedOsmo` succeeding but `DelegateToValidatorSet` failing thus state being reverted?

Reply:
So i did some testing around reverting multi tx calls. It wasn't working as expected via unit test but it worked as expected when i ran cli commands. Nevertheless, for extra security i added a clause so that you need to have existing delegation (nonvalset or valset) to run this msg